agent/lib/internal_address_source.js

```diff
@@ -26,16 +26,9 @@ function extract_spec(def) {
     if (def.spec === undefined) {
         console.error('no spec found on %j', def);
     }
-    var o = myutils.merge(def.spec, {status:def.status});
+    var o = myutils.merge(myutils.merge({}, def.spec), {status:def.status});
```

Comment:
Just fyi, you can call merge with an arbitrary number of arguments, so you could repalce the two calls with two args with one call of three args if you want here. (Not that it is wrong as is)